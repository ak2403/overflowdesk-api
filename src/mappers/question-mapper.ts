import { Mapper } from "../types/mappers/mapper";
import { Question } from "../types/mappers/question";

export class QuestionMapper implements Mapper<Question> {
  private _id!: number;
  private _body!: string;
  private _createdDate!: number;
  private _downVoteCount!: number;
  private _isAnswered: boolean = false;
  private _lastActivityDate!: number;
  private _link!: string;
  private _ownerId!: number;
  private _score!: number;
  private _title!: string;
  private _upVoteCount!: number;
  private _viewCount!: number;

  set id(id: number) {
    this._id = id;
  }

  set body(body: string) {
    this._body = body;
  }

  set createdDate(date: number) {
    this._createdDate = date;
  }

  set downVoteCount(count: number) {
    this._downVoteCount = count;
  }

  set isAnswered(answered: boolean) {
    this._isAnswered = answered;
  }

  set lastActivityDate(date: number) {
    this._lastActivityDate = date;
  }

  set link(link: string) {
    this._link = link;
  }

  set ownerId(id: number) {
    this._ownerId = id;
  }

  set score(score: number) {
    this._score = score;
  }

  set title(title: string) {
    this._title = title;
  }

  set upVoteCount(count: number) {
    this._upVoteCount = count;
  }

  set viewCount(count: number) {
    this._viewCount = count;
  }

  //TODO: throw error if necessary props are missing
  build(): Question {
    return {
      id: this._id,
      body: this._body,
      createdDate: this._createdDate,
      downVoteCount: this._downVoteCount,
      isAnswered: this._isAnswered,
      lastActivityDate: this._lastActivityDate,
      link: this._link,
      ownerId: this._ownerId,
      score: this._score,
      title: this._title,
      upVoteCount: this._upVoteCount,
      viewCount: this._viewCount,
    };
  }
}
