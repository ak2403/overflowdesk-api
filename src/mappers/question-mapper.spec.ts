import { QuestionMapper } from "./question-mapper";

describe("QuestionMapper()", () => {
  const questionMapper = new QuestionMapper();

  it("construct question for passed attribute values", () => {
    questionMapper.id = 1;
    questionMapper.body = "sample-body";
    questionMapper.link = "sample-link";

    expect(questionMapper.build()).toMatchSnapshot();
  });
});
