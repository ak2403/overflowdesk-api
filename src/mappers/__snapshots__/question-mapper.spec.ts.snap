// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionMapper() construct question for passed attribute values 1`] = `
{
  "body": "sample-body",
  "createdDate": undefined,
  "downVoteCount": undefined,
  "id": 1,
  "isAnswered": false,
  "lastActivityDate": undefined,
  "link": "sample-link",
  "ownerId": undefined,
  "score": undefined,
  "title": undefined,
  "upVoteCount": undefined,
  "viewCount": undefined,
}
`;
